Improve modal styles for mobile

unarist opened the pull request:
* Add max-width / max-height to make cancelling by touch backdrop easier
* report: Make p tag in toots inline for `text-overflow: ellipsis`, and reduce margins
* boost: Fix action-bar layout, and remove `overflow-x: scroll`
* actions: Add `:active` style for touching feedback

Update: Added more fixes for landscape view.

* report: Move report comment into action-bar (screenshot without this change is [available here](https://user-images.githubusercontent.com/705555/28729054-031582de-7406-11e7-98eb-63e65bd4433e.png))
**Currently, this change will also applied to PC layout.**
* actions: Make scrollable if needed

cc @sorin-davidoi 

|Before|After|
|-------|------|
|![image](https://user-images.githubusercontent.com/705555/28729045-fdc01c86-7405-11e7-94e1-8d8294104609.png)|![image](https://user-images.githubusercontent.com/705555/28744856-74ad0882-74a5-11e7-9358-9a92c9a14d18.png)|
|![image](https://user-images.githubusercontent.com/705555/28744993-d0e4c574-74a8-11e7-8f2f-e3a928e1abae.png)|![image](https://user-images.githubusercontent.com/705555/28744988-aba53fbe-74a8-11e7-9321-75429af68204.png)|
|![image](https://user-images.githubusercontent.com/705555/28729062-0a2da1b4-7406-11e7-9d94-20a460fb3a66.png)|![image](https://user-images.githubusercontent.com/705555/28729069-0fa5ac4a-7406-11e7-83c4-3d1498d61c64.png)|

sorin-davidoi commented:
Sometimes the report button is not visible (you have to scroll to see it)?

![image](https://user-images.githubusercontent.com/2109702/28731660-14ad71f4-73d5-11e7-85c2-fbe23a756805.png)

unarist commented:
Ah, that's right. Also I cannot see some items on actions modal...

![image](https://user-images.githubusercontent.com/705555/28731839-7ba71ea6-7410-11e7-84fa-459c2eefd02e.png)

unarist commented:
@sorin-davidoi How about this?

![image](https://user-images.githubusercontent.com/705555/28732891-9f863312-7414-11e7-8411-c9899bc4d11e.png)

![image](https://user-images.githubusercontent.com/705555/28732882-93e736a0-7414-11e7-8bec-6eab9abbcc65.png)

and for actions-modal...well, two scroll bars.

![image](https://user-images.githubusercontent.com/705555/28732924-c6712c5c-7414-11e7-9f1d-698006563d94.png)

sorin-davidoi commented:
I only get the scrollbar if I take out `flex-shrink: 0` from `.actions-modal ul`:
![image](https://user-images.githubusercontent.com/2109702/28733172-2339e1de-73db-11e7-93cd-aa58d6b474fb.png)

unarist commented:
I've updated to toggle scrollbar around 430px (I'm not sure this value, but iPhone 5s portrait doesn't need scrollbar I think.)

unarist commented on app/javascript/styles/components.scss:
```diff
@@ -3507,17 +3504,22 @@ button.icon-button.active i.fa-retweet {
 
+    @media screen and (max-height: 430px) {
+      flex-shrink: 1;
+      overflow-y: auto;
```
I thought this style can be set always (outside of media query), but it doesn't show scrollbar on portrait -> landscape changing with that way. So I will set this only if needed.

unarist commented:
@Gargron This improves usability, but also fixed some elements being inaccessible in landscape view. However I have changed some layout, so I'd like to hear your impressions.

Gargron commented:
I do not like the layout change for the report modal, but otherwise this is fine. I'm sorry for not reviewing this sooner. If we can avoid the report modal change then this is good to merge.

Gargron commented:
This was so long this might no longer be relevant? Closing, if it's still relevant, please reopen (and fix conflicts).
